pkg/test/expectations/expectations.go

```diff
@@ -73,10 +75,24 @@ func ExpectCreatedWithStatus(c client.Client, objects ...client.Object) {
 	}
 }
 
-func ExpectDeleted(c client.Client, objects ...client.Object) {
-	for _, object := range objects {
-		Expect(c.Delete(context.Background(), object)).To(Succeed())
+func ExpectDeletedNode(c client.Client, n *v1.Node) {
+	persisted := n.DeepCopy()
+	for index, finalizer := range n.Finalizers {
```

Comment:
IMO you should just set finalizers to []

Reply:
That would make more sense, I'll do that.